language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/sourceprune/LHSSignatureTokenTraverser.java

```diff
@@ -0,0 +1,244 @@
+
+                // Exception is made for capturing explicit object initializations, eg. new Object()
+                boolean explicitInit = prevToken.isPresent() && prevToken.get().getType() == BallerinaParser.NEW;
+
+                // Exception is made for capturing implicit object initializations, eg. Object obj = new()
+                if (!this.captureStatement && this.processedTokens.size() >= 2) {
+                    CommonToken nToken = this.processedTokens.get(this.processedTokens.size() - 1);
+                    CommonToken nnToken = this.processedTokens.get(this.processedTokens.size() - 2);
+                    this.captureStatement = nToken.getType() == BallerinaParser.NEW &&
+                            nnToken.getType() == BallerinaParser.LEFT_PARENTHESIS;
+                }
+                if (!explicitInit && !this.captureStatement && this.pendingLeftParenthesis == 0) {
+                    this.isCapturingEnabled = false;
+                }
+            }
+            if (this.lhsTraverseTerminals.contains(type)) {
+                boolean terminate = terminateLHSTraverse(token.get(), tokenStream);
+                if (terminate) {
+                    break;
+                }
+            }
+            // If match statement, avoid adding semi-colon
+            this.addSemiColon = this.addSemiColon || token.get().getType() == BallerinaParser.MATCH;
+            if (!this.forcedProcessedToken) {
+                processToken(token.get());
+            }
+            this.forcedProcessedToken = false;
+            tokenIndex = token.get().getTokenIndex() - 1;
+            token = tokenIndex < 0 ? Optional.empty() : Optional.of(tokenStream.get(tokenIndex));
+        }
+        sourcePruneContext.put(SourcePruneKeys.LEFT_PARAN_COUNT_KEY, this.pendingLeftParenthesis);
+        sourcePruneContext.put(SourcePruneKeys.RIGHT_PARAN_COUNT_KEY, this.pendingRightParenthesis);
+        sourcePruneContext.put(SourcePruneKeys.LEFT_BRACE_COUNT_KEY, this.pendingLeftBrace);
+        sourcePruneContext.put(SourcePruneKeys.ADD_SEMICOLON_COUNT_KEY, this.addSemiColon);
+        Collections.reverse(this.processedTokens);
+
+        removeProceedingNonDefaultTokens(this.processedTokens);
+        return this.processedTokens;
+    }
+
+    private void removeProceedingNonDefaultTokens(List<CommonToken> processedTokens) {
+        int len = processedTokens.size();
+        for (int i = 0; i < len; i++) {
+            CommonToken token = processedTokens.get(i);
+            if (token.getChannel() != 0) {
+                processedTokens.remove(token);
+                i--;
+                len--;
+            } else {
+                break;
+            }
+        }
+    }
+
+    private boolean terminateLHSTraverse(Token token, TokenStream tokenStream) {
+        int type = token.getType();
+        String text = token.getText();
+        if (type == BallerinaParser.RIGHT_PARENTHESIS) {
```

Comment:
Let's use switch case instead of the if-else ladder?

Reply:
I already tried it but couldn't due to [this](https://github.com/ballerina-platform/ballerina-lang/pull/20524/files#diff-dcda0a7b3b30a1e9cf92ccc9be0904dcR182) case. Even making `${".equals(text)` check an inner-if, didn't work due to false checkstyle error `Fall through from previous branch of the switch statement`.